curator-client/src/main/java/org/apache/curator/ConnectionState.java

```diff
@@ -283,12 +300,19 @@ private boolean checkState(Event.KeeperState state, boolean wasConnected)
             new EventTrace(state.toString(), tracer.get(), getSessionId()).commit();
         }
 
-        if ( checkNewConnectionString && zooKeeper.hasNewConnectionString() )
+        return isConnected;
+    }
+
+    private void handleState(Event.KeeperState state)
+    {
+        if (state == Event.KeeperState.Expired)
+        {
+            handleExpiredSession();
```

Comment:
We've lost functionality here right? If there's a new connection string and an expired session we don't get the new connection string.